arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/raise/RaiseAccumulate.kt

```diff
@@ -547,10 +559,26 @@ public inline fun <Error, A> Raise<NonEmptyList<Error>>.forEachAccumulating(
 public inline fun <Error, A> Raise<NonEmptyList<Error>>.forEachAccumulating(
   iterator: Iterator<A>,
   @BuilderInference block: RaiseAccumulate<Error>.(A) -> Unit
+): Unit = forEachAccumulating(iterator, block, block)
+
+/**
+ * Allows to change what to do once the first error is raised.
+ * Used to provide more performant [mapOrAccumulate].
+ */
+@PublishedApi @JvmName("forEachAccumulatingDouble")
+internal inline fun <Error, A> Raise<NonEmptyList<Error>>.forEachAccumulating(
+  iterator: Iterator<A>,
+  @BuilderInference blockUntilError: RaiseAccumulate<Error>.(A) -> Unit,
+  @BuilderInference blockAfterError: RaiseAccumulate<Error>.(A) -> Unit
```

Comment:
**Nit:** Do we really need two lambdas here? 🤔

Or was that for `JvmName`? Otherwise I am also fine using `forEachAccumulatingImpl` or something. I've started doing that in my own projects, similar how to KotlinX Coroutines does this with `produceImpl`. `flattenMergeImpl`, etc.

Reply:
Why not have:
```
  @BuilderInference block: RaiseAccumulate<Error>.(A) -> B,
  add: (A, B) -> Unit
```
and simply we only call `add` if we don't have any errors.
We could also just have a `Boolean` parameter instead, with similar effects.